tools/node.d.ts

```diff
@@ -204,4 +210,915 @@ export class Dictionary {
+    drop_side_effect_free: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_DefaultAssign extends AST_Binary {
+    optimize: Function;
+}
+
+declare class AST_Conditional extends AST_Node {
+    _walk: Function;
+    transform: Function;
+    needs_parens: Function;
+    _codegen: Function;
+    reduce_vars: Function;
+    _dot_throw: Function;
+    is_boolean: Function;
+    is_number: Function;
+    is_string: Function;
+    _eval: Function;
+    negate: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    drop_side_effect_free: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Array extends AST_Node {
+    _walk: Function;
+    transform: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    _dot_throw: Function;
+    _eval: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    is_constant_expression: Function;
+    drop_side_effect_free: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Object extends AST_Node {
+    _walk: Function;
+    transform: Function;
+    needs_parens: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    _dot_throw: Function;
+    _eval: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    is_constant_expression: Function;
+    drop_side_effect_free: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_ObjectProperty extends AST_Node {
+    _walk: Function;
+    transform: Function;
+    _print_getter_setter: Function;
+    add_source_map: Function;
+    _dot_throw: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+    is_constant_expression: Function;
+    drop_side_effect_free: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_ObjectKeyVal extends AST_ObjectProperty {
+    _codegen: Function;
+    optimize: Function;
+}
+
+declare class AST_ObjectSetter extends AST_ObjectProperty {
+    _codegen: Function;
+    add_source_map: Function;
+}
+
+declare class AST_ObjectGetter extends AST_ObjectProperty {
+    _codegen: Function;
+    add_source_map: Function;
+    _dot_throw: Function;
+}
+
+declare class AST_ConciseMethod extends AST_ObjectProperty {
+    _codegen: Function;
+    optimize: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_Symbol extends AST_Node {
+    mark_enclosed: Function;
+    reference: Function;
+    unmangleable: Function;
+    unreferenced: Function;
+    definition: Function;
+    global: Function;
+    _do_print: Function;
+    _codegen: Function;
+    add_source_map: Function;
+    fixed_value: Function;
+    to_mozilla_ast: Function;
+}
+
+declare class AST_SymbolDeclaration extends AST_Symbol {
+    _find_defs: Function;
+    has_side_effects: Function;
+    may_throw: Function;
+}
+
+declare class AST_SymbolVar extends AST_SymbolDeclaration {
+
+}
```

Comment:
All empty classes should look like this:
```
 declare class AST_SymbolVar extends AST_SymbolDeclaration {}
```

Reply:
Now that I think about it, I don't think such empty classes can exist since they all need constructors that look like:
```
    constructor(props?: object);
```